packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -0,0 +1,160 @@
+import { Component, ComponentInterface, Prop, h, FunctionalComponent, Event, EventEmitter } from "@stencil/core";
+
+import {
+  AdvancedSelectGroupRedirect,
+  AdvancedSelectOption,
+  AdvancedSelectOptionsOrGroup,
+} from "./advanced-select.models";
+import {
+  AdvancedSelectClickEvent,
+  AdvancedSelectOptionClickEvent,
+  AdvancedSelectRedirectClickEvent,
+} from "./advanced-select.interfaces";
+
+@Component({
+  tag: "dso-advanced-select",
+  styleUrl: "advanced-select.scss",
+  shadow: true,
+})
+export class AdvancedSelect implements ComponentInterface {
+  /**
+   * The options to display in the select.
+   */
+  @Prop()
+  options: AdvancedSelectOptionsOrGroup<any>[] = [];
+
+  /**
+   * The active option. By object reference.
+   */
+  @Prop()
+  active?: AdvancedSelectOption<any>;
+
+  /**
+   * An extra text for the active option. Only visible in the list of options.
+   */
+  @Prop()
+  activeHint?: string;
+
+  /**
+   * The open state of the options list.
+   */
+  @Prop()
+  open: boolean = false;
+
+  /**
+   * Emitted when user clicks the select.
+   */
+  @Event({ bubbles: false })
+  dsoClick!: EventEmitter<AdvancedSelectClickEvent>;
+
+  /**
+   * Emitted when user clicks an option
+   */
+  @Event({ bubbles: false })
+  dsoOptionClick!: EventEmitter<AdvancedSelectOptionClickEvent<any>>;
+
+  /**
+   * Emitted when user clicks a redirect link.
+   */
+  @Event({ bubbles: false })
+  dsoRedirectClick!: EventEmitter<AdvancedSelectRedirectClickEvent>;
+
+  private handleClick = (event: MouseEvent) => {
+    this.dsoClick.emit({ originalEvent: event });
+  };
+
+  private handleOptionClick = (event: MouseEvent, option: AdvancedSelectOption<any>) => {
+    this.dsoOptionClick.emit({ originalEvent: event, option });
+  };
+
+  private handleRedirectClick = (event: MouseEvent, redirect: AdvancedSelectGroupRedirect) => {
+    this.dsoRedirectClick.emit({ originalEvent: event, redirect });
+  };
+
+  render() {
+    return (
+      <div>
+        <button class={{ "active-option": true, open: this.open }} type="button" onClick={this.handleClick}>
```

Comment:
Deze button opent en sluit een menu. Dat moeten we duiden met `aria-expanded` en `aria-controls`.

@hansgrimm Kan jij een voorstel doen? Moeten we nog een accessible name zetten, of gaat het goed?

Reply:
Als ik het zo zie, komt er een knoptekst in. Mits dat zo is en die is duidelijk, dan hebben we een accessible name. Volgens mij is `aria-controls` niet nodig zolang de uitklapper naast de knop zit. dan kan je volstaan met `aria-expanded`. Toch, @timveld ?

Reply:
Heb de `aria-expanded` toegevoegd. Als `aria-controls` nodig is hoor ik het graag.